git_patrol_integration_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for git_patrol."""
+
+import asyncio
+import logging
+import os
+import shutil
+import tempfile
+import time
+import unittest
+
+import asyncpg
+import git_patrol_db
+
+
+DB_CONNECT_ATTEMPTS = 3
+DB_CONNECT_WAIT_SECS = 10
+
+
+class GitPatrolIntegrationTest(unittest.TestCase):
+
+  async def _init_git_repos(bare_dir, clone_dir):
+    """Initialize integration test repositories.
+
+    Set up two repositories for the integration test. One repository is created
+    bare to be served via the git HTTP container. The other repository is a
+    clone for making local changes needed by the test.
+
+    Args:
+      bare_dir: Path to the bare repository.
+      clone_dir: Path to the cloned repository.
+    """
+    proc = await asyncio.create_subprocess_exec(
+        'git', 'init', '--quiet', '--bare', cwd=bare_dir)
+    returncode = await proc.wait()
+    assert returncode == 0
```

Comment:
Why are these assert vs assertEquals?  Is it related to async?  I could not find anything that said it should be different?

Reply:
The class' `assertEquals()` method can't be called from an `@classmethod` context. I wanted to fail the suite if the `git` command failed and fortunately Python provides the built-in assert statement.